src/smc_pyutil/smc_pyutil/smc_open.py

```diff
@@ -50,7 +50,8 @@ def main():
         print "Opens each file (or directory) in the Sagemath Cloud web-based editor from the shell."
         print "If the named file doesn't exist, it is created."
     else:
-        process(sys.argv[1:])
+        # ignore args that have ? or * after glob expansion
+        process([p for p in sys.argv[1:] if re.search('[\?\*]', p) is None])
```

Comment:
I think at that point, it's better to keep the `sys.argv[1:]` and do that filtering up there in the `process` function itself. Then, all the logic parts are kept in one place. Despite that, I'm wondering about more explicitly checking if such a file exists. There are also other shell-expansions, e.g. `{2...3}` or something like that, which aren't covered.